src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -150,4 +155,73 @@ class TimeseriesModelNew {
   }
 }
 
-export default TimeseriesModelNew
\ No newline at end of file
+const timeseriesCalculationPool = createCalculationPool({maxSimultaneous: 4, method: 'stack'})
+
+export interface TimeseriesData {
+  getChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => number[]
+  requestChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => void
+  numChannels: () => number
+  numTimepoints: () => number
+  getSampleRate: () => number
+  onDataSegmentSet: (handler: (ds_factor: number, t1: number, t2: number) => void) => void
+}
+
+const useTimeseriesData = (recordingObject: any, recordingInfo: RecordingInfo, hither: HitherContext): TimeseriesData | null => {
+  const ref = useRef<{
+      recordingObject: any | null,
+      recordingInfo: RecordingInfo | null,
+      model: TimeseriesModelNew | null
+  }>({recordingObject: null, recordingInfo: null, model: null})
+  const x = ref.current
+  if ((!x.model) || (x.recordingObject !== recordingObject) || (x.recordingInfo !== recordingInfo)) {
+      const segment_size_times_num_channels = 100000
+      const num_channels = recordingInfo.channel_ids.length
+      const segment_size = Math.ceil(segment_size_times_num_channels / num_channels)
+      const model = new TimeseriesModelNew({
+          samplerate: recordingInfo.sampling_frequency,
+          num_channels,
+          num_timepoints: recordingInfo.num_frames,
+          segment_size
+      })
+      model.onRequestDataSegment((ds_factor: number, segment_num: number) => {
+          ;(async () => {
```

Comment:
What's the semicolon do?

Reply:
It's just a good pattern to add a ";" to lines that start with "(". That's because if the previous line ends with an identifier (and not a semicolon), you will get a hard-to-decipher syntax error.